### PR TITLE
Build User feed

### PR DESCRIPTION
**Pull request by ishitakeshawani**
- Sort posts by date and trend
- added circular progress bars
- used toasts to show errors
- fixed login and signup
- preview link: https://deploy-preview-5--todayspost.netlify.app/

### CODE REVIEW

File: src/components/postcard/PostCard.js [hunk 1]
```diff
@@ -16,12 +16,13 @@ import { Link, useNavigate } from "react-router-dom";
...
 export function PostCard({ post }) {
   const [showModal, setShowModal] = useState(false);
-  const { isPostInEditMode, bookmarks } = usePosts();
+  const { isPostInEditMode, bookmarks, isLikePostLoading } = usePosts();
   const { user } = useAuth();
   const navigate = useNavigate();
   const userName = user?.username;
   const userId = user?._id;
   const dispatch = useDispatch();
+  console.log(post);
```
Comment (rohanmathur91) [inline src/components/postcard/PostCard.js]: remove the console log

File: src/features/profile/profileSlice.js [hunk 2]
```diff
@@ -34,7 +41,11 @@ export const editUserData = createAsyncThunk(
       } = await axios.post("/api/users/edit", { userData });
       return user;
     } catch (error) {
-      console.log(error.response);
+      if (error.response.status === 404) {
+        toast("Please do signup to edit profile!");
+      } else {
+        toast(error.message);
+      }
```
Comment (rohanmathur91) [inline src/features/profile/profileSlice.js:48]: good error handling

**Review by rohanmathur91 — APPROVED** [review]
- Clean code.
- Good naming conventions.
- Looks good to merge

**Review by sashaboi — APPROVED** [review]
Code is very readable and review friendly, 
Liked the folder structure , it clearly shows how the data is flowing.
Looks good to merge!

File: src/features/auth/authSlice.js [hunk 3]
```diff
@@ -19,8 +21,10 @@ export const signup = createAsyncThunk(
         data: { createdUser, encodedToken },
       } = await axios.post("api/auth/signup", userData);
       axios.defaults.headers.common["authorization"] = encodedToken;
+      toast("Sucessfully made an account!");
       return { createdUser, encodedToken };
     } catch (error) {
+      toast(error.message);
```
Comment (hiteshverma27) [inline src/features/auth/authSlice.js:27]: suggestion - give error toast like "something went wrong while trying to make an account" as it is more understandable to user

**Review by hiteshverma27 — APPROVED** [review]
code looks good
good folder structure 
remove console logs

**Pull request merged**